README.md

```diff
@@ -73,15 +73,236 @@ Here are the results of our trained model.
+[keras-yolo3](https://github.com/qqwweee/keras-yolo3) repository with minor modifications and for SSD models we used the 
+[TensorFlow Object Detection API](https://github.com/tensorflow/models/tree/master/research/object_detection)
+repository.
+Both training scripts accept different labels format, and we needed to convert 
+[Yolo_mark](https://github.com/AlexeyAB/Yolo_mark) annotations to those other formats. To accomplish these
+image augmentation and label conversion tasks we have created a [`data_preparer.py`](utils/data_preparer.py).
+With this script, we easily augmented the [4Tzones Traffic Lights Dataset](https://yadi.sk/d/a1Kr8Wmg0zfa0A),
+which enabled the ability to generalize in our models. 
+More about the [`data_preparer.py`](utils/data_preparer.py) script is in 
+the [Data Preparer Script](#data-preparer-script) section. The resulting annotated augmented dataset,
+which is called "4Tzones Traffic Lights Augmented Dataset" can be downloaded using the link below and 
+is licensed under the 
+[Creative Commons Attribution-ShareAlike 4.0 International (CC BY-SA 4.0)](https://creativecommons.org/licenses/by-sa/4.0/) 
+license.   
+
+|      \* & \*\*         |  4Tzones Traffic Lights Augmented Dataset  |  
+|------------------------|          :------------------------:        |  
+| Link                   | https://yadi.sk/d/q2Yyy9PO2SrMKQ           |  
+| License                | [CC BY-SA 4.0](https://creativecommons.org/licenses/by-sa/4.0/) |  
+| Total TL # of Samples  | 11988                                      |  
+| Red TL # of Samples    | 1998                                       |  
+| Yellow TL # of Samples | 1998                                       |  
+| Green TL # of Samples  | 1998                                       |  
+| No TL # of Samples     | 5994                                       |  
+
+\* Notice that the number of samples of red, yellow, and green traffic lights is equal to each other 
+and the number of samples of images without traffic lights is a triple of that. 
+Such a balancing of the dataset is suggested in the
+[How to improve object detection](https://github.com/AlexeyAB/darknet#how-to-improve-object-detection)
+section of README file from [https://github.com/AlexeyAB/darknet](https://github.com/AlexeyAB/darknet) 
+repository by [AlexeyAB](https://github.com/AlexeyAB).  
+
+\** Notice that you cannot obtain the same level of "balancing" from the 
+[4Tzones Traffic Lights Dataset](https://yadi.sk/d/a1Kr8Wmg0zfa0A) and the use of 
+[`data_preparer.py`](utils/data_preparer.py). The process of creating the 
+[4Tzones Traffic Lights Augmented Dataset](https://yadi.sk/d/q2Yyy9PO2SrMKQ) involved several manual steps. 
+It can be roughly described as follows.
+Initially we had stored labeled images from different ROS bags in different folders. On each dataset, 
+we performed flipping, scaling, and balancing. For balancing the number of samples per class was set to about 2000.
```

Comment:
I don't understand, what was set to 2000?

Reply:
I added a comment to the README: `(the [`data_preparer.py`](utils/data_preparer.py) script has a `--balance N` option which commands the script to
create N sample images per each class though augmentation)`. Does it clarify the the situation?

Reply:
But if you set N to 2000, shouldn't red=yellow=green=2000, and no traffic lights =6000?

Reply:
It should not. It can be around 2000. I will rewrite this section to provide a more detailed explanation. Stay tuned.

Reply:
I expanded the explanation. I am not sure whether this paragraph is more understandable now, but it is certainly more detailed.